### PR TITLE
advancedtls: fixed SNI testing and put SNI functions back in advancedtls.go

### PR DESCRIPTION
**Pull request by cindyxue**
This PR does the following things:
1. Fixed bugs in SNI unit test
2. Moved SNI unit test back to `advancedtls_test.go` and moved SNI function back to `advancedtls.go` after go1.9 is deprecated

@ZhenLian

### CODE REVIEW

**Comment by ZhenLian** [comment]
@dfawley Sorry for letting the tests slip away. This PR should fix them. Could you please take a look? Thanks!

**Comment by cindyxue** [comment]
@ZhenLian Yes, I was comparing a pointer with a certificate and that's why it failed TAT. Forgot that the value type we got from `GetCertificate` is a pointer to the certificate instead of a certificate... 
Thus, this is able to fix the issue: 
`cmp.Equal(*gotCertificate, test.wantCert, cmp.AllowUnexported(big.Int{}))`

File: security/advancedtls/advancedtls_test.go [hunk 1]
```diff
@@ -688,3 +690,81 @@ func (s) TestOptionsConfig(t *testing.T) {
...
+			serverGetCert: func(info *tls.ClientHelloInfo) ([]*tls.Certificate, error) {
+				return []*tls.Certificate{&serverCert1, &serverCert2, &serverCert3}, nil
+			},
+			// "foo.bar.server2.com" is the common name on server certificate server_cert_2.pem.
+			serverName: "foo.bar.server2.com",
+			wantCert:   serverCert2,
+		},
+		{
+			desc: "Select serverCert3",
+			serverGetCert: func(info *tls.ClientHelloInfo) ([]*tls.Certificate, error) {
```
Comment (dfawley) [inline security/advancedtls/advancedtls_test.go:735]: I notice this is the same in all three tests.  Is there any reason to parameterize it?  Do we want to change the order in some tests?

Comment (cindyxue) [reply security/advancedtls/advancedtls_test.go]: You are right. We don't need it as a parameter, so I removed it now :)

**Comment by ZhenLian** [comment]
@dfawley Thank you so much for the review!

**Pull request merged**